lib/app/utils.js

```diff
@@ -19,6 +19,67 @@ export function globalHandleError(error) {
   }
 }
 
+// Takes an array of errors and assigns a message and stack to it to make it error-like:
+export function compactMultipleErrors(errors, messagePrefix = 'Multiple Errors', errorJoiner = '\n + \n'){
```

Comment:
We loose things like error statusCode here. Maybe we can assign a default value of 500 and then override it if any error had more precise one

Reply:
Is there a particular test case or project that is currently throwing an error that contains a `statusCode` property in something other than the `asyncData` or `fetch` methods? Currently, for asyncData rejections, the first (and likely only) error is thrown directly instead. See [`lib/app/server.js:216`](https://github.com/nuxt/nuxt.js/blob/a3ad155b9a595bd92145dbbdf4ca8661baa175dd/lib/app/server.js#L216)&[`:229`](https://github.com/nuxt/nuxt.js/blob/a3ad155b9a595bd92145dbbdf4ca8661baa175dd/lib/app/server.js#L229), [`lib/app/client.js:391`](https://github.com/nuxt/nuxt.js/blob/a3ad155b9a595bd92145dbbdf4ca8661baa175dd/lib/app/client.js#L391)&[`:404`](https://github.com/nuxt/nuxt.js/blob/a3ad155b9a595bd92145dbbdf4ca8661baa175dd/lib/app/client.js#L404).

If you think it makes sense to add a `statusCode` property to the errors array, should it be based on the first truthy `statusCode` property in the errors array (like already done with the `stack` property), or by some other logic (with the default being 500)? I will refactor it assuming the former in a new commit